charmhelpers/contrib/database/mysql.py

```diff
@@ -547,36 +548,48 @@ def get_mem_total(self):
                     mtot, modifier = mem.strip().split(' ')
                     return '%s%s' % (mtot, modifier[0].upper())
 
-    def parse_config(self):
-        """Parse charm configuration and calculate values for config files."""
-        config = config_get()
-        mysql_config = {}
-        if 'max-connections' in config:
-            mysql_config['max_connections'] = config['max-connections']
+    def get_innodb_flush_log_at_trx_commit(self):
+        """Get value for innodb_flush_log_at_trx_commit.
 
-        if 'wait-timeout' in config:
-            mysql_config['wait_timeout'] = config['wait-timeout']
+        Use the innodb-flush-log-at-trx-commit or the tunning-level setting
+        translated by INNODB_FLUSH_CONFIG_VALUES to get the
+        innodb_flush_log_at_trx_commit value.
 
-        if 'innodb-flush-log-at-trx-commit' in config:
-            mysql_config['innodb_flush_log_at_trx_commit'] = \
-                config['innodb-flush-log-at-trx-commit']
-        elif 'tuning-level' in config:
-            mysql_config['innodb_flush_log_at_trx_commit'] = \
-                self.INNODB_FLUSH_CONFIG_VALUES.get(config['tuning-level'], 1)
+        :returns: Numeric value for innodb_flush_log_at_trx_commit
+        :rtype: int
+        """
+        if config_get('innodb-flush-log-at-trx-commit'):
+            return config_get('innodb-flush-log-at-trx-commit')
+        elif config_get('tuning-level'):
+            return self.INNODB_FLUSH_CONFIG_VALUES.get(
+                config_get('tuning-level'), 1)
```

Comment:
This code has a subtle potential bug (it's a partial function);  if neither `if` is truthy then the function returns `None` (silently).

Either this is expected (in which case, `:rtype: Optional[int]` or it should raise an `Exception` for the situation when neither of the conditions is true.

Also, repeated calls to config_get, which could be rationalised to 2 with intermediate vars.

Reply:
Updating the docstring